src/banded/BandedMatrix.jl

```diff
@@ -77,16 +81,16 @@ for MAT in (:AbstractBandedMatrix, :AbstractMatrix, :AbstractArray)
 end
 
 
-#TODO: Add test
-function Base.convert(::Type{BM}, M::Matrix) where {BM<:BandedMatrix}
-    ret = BandedMatrix{eltype(BM) == Any ? eltype(M) :
-                        promote_type(eltype(BM),eltype(M))}(undef, size(M,1),size(M,2),size(M,1)-1,size(M,2)-1)
+function Base.convert(BM::Type{<: BandedMatrix{T}}, M::Matrix) where {T}
```

Comment:
This is also problematic (though the existing function was too) as we have, e.g., `BandedMatrix{T,SparseMatrixCSC{T}} <: BandedMatrix{T}` but the function returns `BandedMatrix{T,Matrix{T}}`. 
I'd suggest changing this to 
```julia
function Base.convert(BM::Type{BandedMatrix{T,V}}, M::Matrix) where V<:AbstractMatrix{T} where T
    ret = BandedMatrix{T,V}(undef, size(M,1),size(M,2),size(M,1)-1,size(M,2)-1)
    for k=1:size(M,1),j=1:size(M,2)
        ret[k,j] = convert(T, M[k,j])
    end
    ret
end

Base.convert(BM::Type{BandedMatrix{T}}, M::Matrix)where T = convert(BandedMatrix{T,Matrix{T}}, M)
```

Reply:
Good point. Actually, there is no reason M should not be an abstract matrix.

Is it correct that the function does not try to figure out upper and lower bounds? In which case, we could simply use `_BandedMatrix`, with `convert(fieldtype(BM, :data), M)` as first argument. That would be more general.

Reply:
OK.